### PR TITLE
#1401. Tests for invocation keys

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Patterns/invocation_keys_A04_t02.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+import "patterns_collections_lib.dart";
+import "../../Utils/expect.dart";
+
+String test1(Object o) {
+  switch (o) {
+    case [0, var x?]: // Expect calls [0], [1]
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Patterns/invocation_keys_A04_t02.dart]: Thank you. Updated

File: LanguageFeatures/Patterns/invocation_keys_A05_t01.dart [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  log += s;
+}
+
+const c1 = C(1, logger);
+const c2 = C(2, logger);
+const c3 = C(3, logger);
+
+String test1(int i) {
+  switch (i) {
+    case const(c1):
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Patterns/invocation_keys_A05_t01.dart]: Changed to `const C(1, logger)`

File: LanguageFeatures/Patterns/invocation_keys_A06_t01.dart [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+/// 2. For each pattern p in s with parent invocation i, bind invocation keys to
+///   it and its subpatterns using the following procedure:
+/// To bind invocation keys in a pattern p using parent invocation i:
+/// ...
+/// Identifier:
+/// i. In a matching context, the same as a constant pattern whose constant
+///   expression is the identifier.
+/// ii. Else, nothing to do.
+///
+/// @description Checks that invocation key `("identifier==", [arg])` is called
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Patterns/invocation_keys_A06_t01.dart:24]: In the previous test now we have `const C(1, logger)`. So this one I leaved as is

**Review by sgrekhov — COMMENTED** [review]
Tests updated. Please take another look

File: LanguageFeatures/Patterns/invocation_keys_A04_t04.dart [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+import "patterns_collections_lib.dart";
+import "../../Utils/expect.dart";
+
+String test1(Object o) {
+  switch (o) {
+    case ([0, 1]): // Expect calls [0], [1]
```
[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Patterns/invocation_keys_A04_t04.dart]: Thanks!

File: LanguageFeatures/Patterns/invocation_keys_A06_t01.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Patterns/invocation_keys_A06_t01.dart:24]: Good, thanks!

**Review by eernstg — APPROVED** [review]
LGTM!

**Pull request merged**